schema/pgsql/schema.sql

```diff
@@ -344,89 +362,108 @@ CREATE TABLE rule_escalation (
+    rule_escalation_id bigint NOT NULL,
+    contact_id bigint,
+    contactgroup_id bigint,
+    schedule_id bigint,
+    channel_id bigint,
 
     changed_at bigint NOT NULL,
     deleted boolenum NOT NULL DEFAULT 'n',
 
     CONSTRAINT pk_rule_escalation_recipient PRIMARY KEY (id),
-
-    CHECK (num_nonnulls(contact_id, contactgroup_id, schedule_id) = 1)
+    CONSTRAINT ck_rule_escalation_recipient_only_one_recipient CHECK (num_nonnulls(contact_id, contactgroup_id, schedule_id) = 1),
+    CONSTRAINT fk_rule_escalation_recipient_rule_escalation FOREIGN KEY (rule_escalation_id) REFERENCES rule_escalation(id),
+    CONSTRAINT fk_rule_escalation_recipient_contact FOREIGN KEY (contact_id) REFERENCES contact(id),
+    CONSTRAINT fk_rule_escalation_recipient_contactgroup FOREIGN KEY (contactgroup_id) REFERENCES contactgroup(id),
+    CONSTRAINT fk_rule_escalation_recipient_schedule FOREIGN KEY (schedule_id) REFERENCES schedule(id),
+    CONSTRAINT fk_rule_escalation_recipient_channel FOREIGN KEY (channel_id) REFERENCES channel(id)
 );
 
 CREATE INDEX idx_rule_escalation_recipient_changed_at ON rule_escalation_recipient(changed_at);
 
 CREATE TABLE incident (
     id bigserial,
-    object_id bytea NOT NULL REFERENCES object(id),
+    object_id bytea NOT NULL,
     started_at bigint NOT NULL,
     recovered_at bigint,
     severity severity NOT NULL,
 
-    CONSTRAINT pk_incident PRIMARY KEY (id)
+    CONSTRAINT pk_incident PRIMARY KEY (id),
+    CONSTRAINT fk_incident_object FOREIGN KEY (object_id) REFERENCES object(id)
 );
 
 CREATE TABLE incident_event (
-    incident_id bigint NOT NULL REFERENCES incident(id),
-    event_id bigint NOT NULL REFERENCES event(id),
+    incident_id bigint NOT NULL,
+    event_id bigint NOT NULL,
 
-    CONSTRAINT pk_incident_event PRIMARY KEY (incident_id, event_id)
+    CONSTRAINT pk_incident_event PRIMARY KEY (incident_id, event_id),
+    CONSTRAINT fk_incident_event_incident FOREIGN KEY (incident_id) REFERENCES incident(id),
+    CONSTRAINT fk_incident_event_event FOREIGN KEY (event_id) REFERENCES event(id)
 );
 
 CREATE TYPE incident_contact_role AS ENUM ('recipient', 'subscriber', 'manager');
 
 CREATE TABLE incident_contact (
-    incident_id bigint NOT NULL REFERENCES incident(id),
-    contact_id bigint REFERENCES contact(id),
-    contactgroup_id bigint REFERENCES contactgroup(id),
-    schedule_id bigint REFERENCES schedule(id),
+    incident_id bigint NOT NULL,
+    contact_id bigint,
+    contactgroup_id bigint,
+    schedule_id bigint,
     role incident_contact_role NOT NULL,
 
-    CONSTRAINT key_incident_contact_contact UNIQUE (incident_id, contact_id),
-    CONSTRAINT key_incident_contact_contactgroup UNIQUE (incident_id, contactgroup_id),
-    CONSTRAINT key_incident_contact_schedule UNIQUE (incident_id, schedule_id),
-    CONSTRAINT nonnulls_incident_recipients_check CHECK (num_nonnulls(contact_id, contactgroup_id, schedule_id) = 1)
+    -- Keep in sync with internal/incident/db_types.go!
```

Comment:
Added that little hint here but forgot to adjust the references in the GO code :).